### PR TITLE
WV-3881 Fix regex for replacing breadcrumbs, main content and title.

### PR DESCRIPTION
**Pull request by maximilianmikus**

### CODE REVIEW

File: src/ResponseMerger/NuxtResponseMerger.php [hunk 1]
```diff
@@ -29,8 +29,8 @@ public function mergeResponses(ResponseInterface $backendResponse, ResponseInter
 
     // Finally, merge responses and serve them.
     $page = $frontendResponse->getBody()->__toString();
-    $page = preg_replace('/<title(.*)>*<\/title>/', '<title$1>' . strip_tags($data['title']) . '</title>', $page);
-    $page = preg_replace('/<main role="main"(.*)><\/main>/', '<main role="main"$1>' . $data['content'] . '</main>', $page);
+    $page = preg_replace('/<title((?:\s*[a-zA-Z\-0-9]+(?:="[^"]*")*\s*)*)>[^<]*<\/title>/', '<title$1>' . strip_tags($data['title']) . '</title>', $page);
```
Comment (fago) [inline src/ResponseMerger/NuxtResponseMerger.php]: please make this more clear what this goes for. e.g. add a class constant ATTRIBUTE_REGEX for the first part

File: src/ResponseMerger/NuxtResponseMerger.php [hunk 2]
```diff
@@ -29,8 +29,8 @@ public function mergeResponses(ResponseInterface $backendResponse, ResponseInter
...
+    $page = preg_replace('/<main role="main"((?:\s*[a-zA-Z\-0-9]+(?:="[^"]*")*\s*)*)><!----><\/main>/', '<main role="main"$1>' . $data['content'] . '</main>', $page);
```
Comment (fago) [inline src/ResponseMerger/NuxtResponseMerger.php]: This makes it now require a comment, what sucks:
> 
the comment must be optional, we really should not require something optional like a comment

**Review by fago — CHANGES_REQUESTED** [review]

File: src/ResponseMerger/NuxtResponseMerger.php (hunk 2 above)
Comment (maximilianmikus) [reply src/ResponseMerger/NuxtResponseMerger.php]: hmm ok I thought this was static.. np I will just use the solution as for title

File: src/ResponseMerger/NuxtResponseMerger.php [hunk 3]
```diff
@@ -29,8 +34,9 @@ public function mergeResponses(ResponseInterface $backendResponse, ResponseInter
 
     // Finally, merge responses and serve them.
     $page = $frontendResponse->getBody()->__toString();
-    $page = preg_replace('/<title(.*)>*<\/title>/', '<title$1>' . strip_tags($data['title']) . '</title>', $page);
-    $page = preg_replace('/<main role="main"(.*)><\/main>/', '<main role="main"$1>' . $data['content'] . '</main>', $page);
+    $attribute_regex = self::ATTRIBUTE_REGEX;
+    $page = preg_replace("/<title$attribute_regex>.*<\/title>/", '<title$1>' . strip_tags($data['title']) . '</title>', $page);
```
Comment (fago) [inline src/ResponseMerger/NuxtResponseMerger.php]: thx, way better. small remark here still, could we leave the outer () out of $attribute_regex. Then reading this line becomes easy - sicne it's important to see what $1 is matching.

**Review by fago — CHANGES_REQUESTED** [review]

**Review by fago — APPROVED** [review]

**Pull request merged**